crates/sui-types/src/error.rs

```diff
@@ -59,42 +59,142 @@ macro_rules! exit_main {
     };
 }
 
+#[derive(
+    Eq, PartialEq, Clone, Debug, Serialize, Deserialize, Error, Hash, AsRefStr, IntoStaticStr,
+)]
+pub enum UserInputError {
```

Comment:
Thoughts on the naming here? Would something like `TransactionInputError` be a bit more precise? 

We have various types of user inputs, and this doesn't cover all of them. For example, the user specifies things for call arguments for Move functions, but those are in ExecutionErrors, not here even though they are "input"s. Just raising the potential confusion here

Reply:
I thought about `TransactionInputError` that. `UserInputError` stands out because the error could also come from the read path, e.g. get object etc. So `TransactionInput` could be a bit misleading.

There are currently a few `ExecutionErrors` raised during the signing phase which we should fix (this PR fixes a couple). Btw, i think `the user specifies things for call arguments for Move functions` is also considered "user inputs"